homeassistant/components/swiss_public_transport/config_flow.py

```diff
@@ -0,0 +1,82 @@
+"""Config flow for transport.opendata.ch."""
+import logging
+from typing import Any
+
+from opendata_transport import OpendataTransport
+from opendata_transport.exceptions import OpendataTransportError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_DESTINATION, CONF_START, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_START): cv.string,
+        vol.Required(CONF_DESTINATION): cv.string,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SwissPublicTransportConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Swiss public transport config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Async user step to set up the connection."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            session = async_get_clientsession(self.hass)
+            opendata = OpendataTransport(
+                user_input[CONF_START], user_input[CONF_DESTINATION], session
+            )
+            try:
+                await opendata.async_get_data()
+            except OpendataTransportError:
+                errors["base"] = "client"
```

Comment:
I would prefer to be named "cannot_connect" to keep in line with other integrations as "client" doesn't really say what's up